packages/dcos-integration-test/extra/test_3dt.py

```diff
@@ -458,28 +458,63 @@ def test_3dt_bundle_create(cluster):
     # make sure the job is done, timeout is 5 sec, wait between retying is 1 sec
     status_url = '/system/health/v1/report/diagnostics/status/all'
 
-    # the job executing takes about 1 seconds per node. A static timeout does not work very well for every context
-    # this test can be run in. Let's calculate a timeout based on how many nodes in a cluster + 15% tolerance.
-    # the timeout would looks like: (len(all_nodes) + 15%) * 1000
-    timeout = len(cluster.masters + cluster.all_slaves) * 1000
-    tolerance = int(15 * timeout / 100.0)
-    job_timeout = timeout + tolerance
-    logging.info("Using timeout {}".format(job_timeout))
-
-    @retrying.retry(stop_max_delay=job_timeout, wait_fixed=1000)
+    # lets make sure we are progressing. We need at least 2 points to compare.
+    def is_job_progressing(job_progress):
```

Comment:
First time we run through this if 3dt is taking some time, wouldn't this hit the "not 2 data points" bit which we don't want it to since we want to fail hard if we're not making progress?